### ISSUE TITLE
Support tdlib > 1.8.5

### ISSUE THREAD

**Issue by selankon** [alexander-akhmetov/python-telegram#348]
Hi, 

This issue is directly related to https://github.com/tdlib/td/issues/2285. 

I had a working client with a valid api_id and everything working in my computer, but I bought a new one, so on the moving I tried to move also this project.

I moved the project itself, installed the dependencies, and build the tdlib on the new computer (https://github.com/tdlib/td/commit/d912fd14311fec0d5d6b81359a4bc7d9ce2b0996). Also I moved the tdlib files folder.

But when I try to run my client, with the original client stopped, I got this error:

```
Telegram error: {'@type': 'error', 'code': 400, 'message': 'Valid api_id must be provided. Can be obtained at https://my.telegram.org', '@extra': {'request_id': 'updateAuthorizationState'}}
```

They said that probably I'm using a different tdlib version, and ask me to use `getOption("version")`. 

However, I can't really understand how to do this (because I don't understand how the wrapper really works).

Could you kindly point me how to do this call? I found the `_send_data` function on the wrapper, probably I can use something similar to do a call against tdlib asking for the version? 

Thanks in advance

**Comment by alexander-akhmetov** [comment]
Hi! Thanks for the link to the issue. `python-telegram` indeed does not support the new tdlib API. It looks like at least the `setTdlibParameters` call [here](https://github.com/alexander-akhmetov/python-telegram/blob/061a3a1ff16e0aafcc06b3cca192ecf012fc8cf3/telegram/client.py#L729) should be changed based on the tdlib version used by the wrapper. I'll adapt the wrapper, but in the meantime you can build tdlib 1.8.5 (https://github.com/tdlib/td/commit/d9cfcf88fe4ad06dae1716ce8f66bbeb7f9491d9), it should work with the current implementation.

> I found the _send_data function on the wrapper, probably I can use something similar to do a call against tdlib asking for the version?

To get the version you could do something like this:

```
tg = Telegram(
    api_id=...,
    api_hash=...,
    phone=...,
    database_encryption_key=...,
)

result = tg.call_method('getOption', params={'name': 'version'})
result.wait()
print(result.update)
```

**Comment by selankon** [comment]
Ok, thanks for the answer! 

The commit you are using is really back between the one that I'm using that is working too on the first computer. Anyway, I will compile the 1.8.5 version. 

And thanks to show me how to use the wrapper to do tdlib calls.

**Comment by selankon** [comment]
It is still strange, on the computer with the working version appear to be using tdlib 1.8.5, meanwhile, the commit on my tdlib folder is above 1.8.10. Which mean that I'm using another tdlib version that the one I compiled.. 

Looking for tdlib files on my system I don't found  anymore that the ones I compiled... Strange

**Comment by alexander-akhmetov** [comment]
If you don't specify `library_path` when you are instantiating the client (`Telegram(library_path=...)`), `python-telegram` [tries to find it](https://github.com/alexander-akhmetov/python-telegram/blob/061a3a1ff16e0aafcc06b3cca192ecf012fc8cf3/telegram/tdjson.py#L13) and if it's unable to find the system library it uses its own version (`1.8.5` currently). 

[It should log the path to the library that its using](https://github.com/alexander-akhmetov/python-telegram/blob/061a3a1ff16e0aafcc06b3cca192ecf012fc8cf3/telegram/tdjson.py#L31), in the logs you should see something like this :
```
2023-02-01 10:22 [INFO] telegram.tdjson: Using shared library "/path/to/libtdjson"
```

**Comment by selankon** [comment]
I don't see any log. And I thought that is mandatory to build tdlib by your own, I didn't know that `python_telegram` has its own tdlib.

**Comment by alexander-akhmetov** [comment]
You can try to also debug it by calling the function that python-telegram uses to find the library:

```shell
$ python3

>>> from telegram.tdjson import _get_tdjson_lib_path
>>> print(_get_tdjson_lib_path())
/path/to/libtdjson
```

Or set manually the exact path to the version that you want to use with `library_path`:

```python
tg = Telegram(
    api_id=...,
    api_hash=...,
    phone=...,
    database_encryption_key=...,
    ...
    library_path='/path/to/libtdjson',
)
```

**Comment by selankon** [comment]
Thats nice thanks!

**Comment by Sakura0721** [comment]
I met the same problem. Hope python-telegram can be adapted to tdlib 1.8.11!

**Comment by Hi-Angel** [comment]
Does it currently work with `1.8.10` ? I'm trying to run a "hello world" from the project documentation, but get the same error about api_id being invalid. I presume it's this bug? Should it be reopened?

**Comment by alexander-akhmetov** [comment]
Hi, no, it does not work with the new tdlib. Its APIs have changed and the library needs to be adapted to support it.

**Comment by alexander-akhmetov** [comment]
python-telegram `0.18.0` is released and it works with tdlib > 1.8.5 (thanks to @JleMyP )
